version: "3.8"

services:
  homeassistant:    
    image: "ghcr.io/home-assistant/home-assistant:stable"
    container_name: home-server
    network_mode: host
    environment:
      VIRTUAL_HOST: "${DOMAIN}"
      VIRTUAL_PORT: 8123
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /run/dbus:/run/dbus:ro
      - ./config:/config      
    privileged: true
  matter-server:
    image: ghcr.io/home-assistant-libs/python-matter-server:stable
    container_name: home-matter
    network_mode: host
    environment:
      NETWORK_ACCESS: internal
    security_opt:
      - apparmor=unconfined
    volumes:
      - /run/dbus:/run/dbus:ro
      - ./matter:/data
  multipan:
    image: b2un0/silabs-multipan-docker:latest
    container_name: home-thread
    network_mode: host
    privileged: true
    volumes:
      - ./thread:/data
    environment:
      DEVICE: "/dev/ttyUSB0"
      FLOW_CONTROL: "true"
      BAUDRATE: "460800"
      BACKBONE_IF: "eth0"
      NETWORK_ACCESS: internal
      OTBR_REST_LISTEN_PORT: "8081"
      OTBR_WEB_PORT: "8086"
      EZSP_LISTEN_PORT: "20108" #
      AUTOFLASH_FIRMWARE: 1 # change to 1 for firmware update
      FIRMWARE: "NabuCasa_SkyConnect_RCP_v4.3.1_rcp-uart-hw-802154_460800.gbl"
